assets/javascripts/validation/form.js

```diff
@@ -42,142 +46,149 @@ Inline error markup example:
-      </a>
-    </li>
-    <li class="hidden" role="tooltip" data-journey="search-page:error:country-code-auto-complete">
-      <a href="#country-code-auto-complete" class="error-list" data-focuses="country-code-auto-complete" id="country-code-auto-complete-error">
-        Error Message Auto Complete
-      </a>
-    </li>
- </ul>
-
- @* ----- Form Summary Server side errors ----- *@
- @if(registrationForm.hasErrors){
-    <ul class="js-error-summary-messages">
-        @registrationForm.errors.map { error =>
-            <li role="tooltip" data-journey="search-page:error:@error.key">
-                <a href="#@error.key" class="error-list" data-focuses="@error.key" id="@{error.key}-error">
-                @Messages(error.message)
-                </a>
-            </li>
-        }
-    </ul>
-  }
+ <div class="flash error-summary@if(form.hasErrors) { error-summary--show}"
+      id="Search-failure"
+      role="group"
+      aria-labelledby="errorSummaryHeading"
+      tabindex="-1">
+
+      <h2 id="errorSummaryHeading" class="h3-heading">@heading</h2>
+      <ul class="js-error-summary-messages">
+      @if(form.hasErrors) {
+        @form.errors.map(error => fragments.errorSummaryItem(error.key, error.message))
+      }
+      </ul>
  </div>
  */
 
 var $forms;
 
-var displayErrorSummary = function (validator, $errorSummary) {
-  var $errorSummaryMessages = $errorSummary.find('.js-error-summary-messages > li');
-  var invalidInputs = validator.invalid;
-
-  $errorSummaryMessages.each(function (index, elem) {
-    var $elem = $(elem);
-    var identSplit = $elem.data('journey').split('search-page:error:');
-    var validationId = identSplit && identSplit[1];
-
-    // find the error message and display
-    if (invalidInputs[validationId]) {
-      //do not re-show server-rendered error messages when performing JS validation
-      if ($elem.hasClass('server-rendered')) {
-        $elem.addClass('hidden');
-      } else {
-        $elem.removeClass('hidden');
-      }
-    } else {
-      $elem.addClass('hidden');
-    }
-  });
-
-  $errorSummary.removeClass('hidden');
-};
-
-var purgeErrorList = function (errorList, $elem) {
-  for (var i = errorList.length; i-- > 0; ) {
-    var errorDetail = errorList[i];
-    if (errorDetail.element.name === $elem[0].name) {
-      errorList.splice(i, 1);
-    }
+var displayGlobalErrorSummary = function (validator, errorMessages) {
+  if (errorMessages.length) {
+    $(errorMessages).each(function (index, errorDetail) {
+      createErrorSummaryListItem(validator, errorDetail.message, errorDetail.name);
+    });
   }
 };
 
-var flushErrors = function (invalid, errorList) {
-  for(var inputName in invalid) {
+var createErrorSummaryListItem = function (validator, error, name) {
+  var $errorSummaryMessages = $(validator.currentForm).find('.js-error-summary-messages');
+  var $template = $('<li role="tooltip"><a></a></li>');
+  var $errorSummaryListElement = $template.clone();
```

Comment:
Is this clone necessary? As far as I know the line 93 should create the new DOM element and there should not be need to make a clone of it.